### PR TITLE
feat: add esbuild option for typescript

### PR DESCRIPTION
**Pull request by wattanx**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Migrate the transpile of typescript to `esbuild`.
I decided to opt-in because I wanted to test for a while to see if there were any problems.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/bridge/src/webpack/esbuild.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import type { Nuxt } from '@nuxt/schema'
+
+export interface WebpackConfigContext {
+  nuxt: Nuxt
+  config: Configuration
+  isServer: boolean
+  transpile: RegExp[]
+}
+
+export function esbuild (ctx: WebpackConfigContext) {
```
Comment (wattanx) [inline packages/bridge/src/webpack/esbuild.ts:12]: Based on the following files:
https://github.com/nuxt/nuxt/blob/main/packages/webpack/src/presets/esbuild.ts

**Comment by wattanx** [comment]
After https://github.com/nuxt/bridge/pull/1064 is merged, I would like to add this for minor releases.
(as it may have a large impact).

**Comment by wattanx** [comment]
Because this is a big change, I try to opt-in at first.

**Comment by wattanx** [comment]
I added it!
What do you think? Is it a little hard to understand?
https://github.com/nuxt/bridge/pull/1065/commits/bc969d01961f070258ec04cb4170959963a4c73e

File: test/bridge.test.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { expectNoClientErrors, parseData, renderPage } from './utils'
 
 const isWebpack = process.env.TEST_BUILDER === 'webpack'
 const isDev = process.env.TEST_ENV === 'dev'
+const isTSX = process.env.TEST_TYPESCRIPT === 'isTSX' || process.env.TEST_TYPESCRIPT !== 'esbuild'
```
Comment (danielroe) [inline test/bridge.test.ts:8]: do we not expect TSX to work with `esbuild`?

Or maybe we should always set `isTSX` and only use matrix for testing esbuild enabled/not?

**Pull request merged**